utils/utils.hpp

```diff
@@ -0,0 +1,73 @@
+/**
+ * @file utils.hpp
+ * @author Kartik Dutt
+ * 
+ * Definition of Utils class.
+ * 
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MODELS_UTILS_HPP
+#define MODELS_UTILS_HPP
+
+#include <iostream>
+#include <cstdlib>
+#include <sys/stat.h>
+#include <openssl/sha.h>
```

Comment:
If we need OpenSSL, I think that's just fine, but should we add it to the dependencies we search for and require in the main `CMakeLists.txt`?

Reply:
I am going a different way at least for now, I am writing my own implementation of checksum.